developers/tutorials/ipfs-filecoin/using-nftstorage.md

```diff
@@ -0,0 +1,451 @@
+   ```
+
+5. **Deploy NFT contract to Harmony Testnet**
+
+   In order to deploy the NFT smart contract to Harmony Testnet, you need to let Truffle know what is the right network configuration and which harmony wallet to use. All those could be done by configuring truffle in `truffle-config.js`.
+
+   First, install `@truffle/hdwallet-provider` via npm first.
+
+   ```shell
+   npm install --save-dev @truffle/hdwallet-provider
+   ```
+
+   Add harmony testnet configuration to `truffle-config.js` file. Also make sure to request some ONE token in the wallet so we can pay for transaction fees for smart contract deployment.
+
+   ```javascript
+   const HDWalletProvider = require('@truffle/hdwallet-provider');
+   const privateKeyTest = '<ADD-YOUR-PRIVATE-KEY-HERE>';//If you have harmony wallet in MetaMask, you can export the private key.
+   
+   ....<other code>....
+   networks: {
+       harmonyTestnet: {
+         provider: () => {
+           if (!privateKeyTest.trim()) {
+             throw new Error(
+               'Please enter a private key with funds, you can use the default one'
+             );
+           }
+           return new HDWalletProvider({
+             privateKeys: [privateKeyTest],
+             providerOrUrl: 'https://api.s0.b.hmny.io',
+           });
+         },
+         network_id: 1666700000,
+       },
+     },
+   ... <other code>...
+   ```
+
+   Now the truffle configuration is all set, wen can run the following command to deploy the NFT contract to Harmony Testnet.
+
+   ```shell
+   truffle deploy --network harmonyTestnet
+   ```
+
+   Once the smart contract is deployed success, remember to record your **smart contract address** which we will also use it later. You can also find your NFT contract on [Harmony Testnet blockchain explorer](https://explorer.pops.one/hrc721), also test it on [Remix](https://remix.ethereum.org/).
+
+### Connect to Metamask 
+
+In this step, we will write some front-end code to connect to MetaMask wallet so that you can use is to mint NFT and pay transaction fees in next step. 
+
+Before we move forward, make sure we have installed MetaMask browser extension and add Harmony Network RPC Endpoint. Check more instructions [here](https://docs.harmony.one/home/network/wallets/browser-extensions-wallets/metamask-wallet#1.-adding-a-custom-rpc-endpoint).
+
+Now, let's start coding. 
+
+1. **Create a WalletConnect component**
+
+   We have already created the react app from the first step, so we can directly adding our code into `harmony-nft/src` folder. 
+
+   Create a new folder call components under `src` folder, then create a `WalletConnect.js` file which we will add our code to detect wallet provider and connect to MetaMask.   [The full WalletConnect.js file can be found on Github](https://github.com/longfeiWan9/harmony-nft-demo/blob/master/src/components/WalletConnect.js).
+
+   Also make sure requesting some ONE token in the wallet so we can pay for transaction fees for minting NFT.
```

Comment:
clarify please

Reply:
harmony's token is $ONE... makes for confusing instructions. i made a suggestion to clarify.

Reply:
Thank you Mosh for explaining.